Assignment Link Validation for LinkIDs

maleekha opened the pull request:
**Changes Made**

- Changed the linkdto id datatype from `string` to `Guid`
- Added error handling in the SAVE and GET Api calls for cases where the incoming request or the storage contains a malformed linkID which cannot be parsed to `Guid`
- Added UI changes for when certain linkIDs are invalid and are not being rendered

adbindal commented on backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs:
```diff
@@ -104,9 +107,16 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
+            }
 
-            if (linkId != linkDto.Id)
+            if (linkId != linkDto.Id.ToString())
```
What would it take to start accepting `linkId` as **Guid** also?

maleekha replied:
The row key in an azure table can only be a string; we cannot override this explicitly, which is why we are having to put this validation in the dto.

adbindal commented on backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs:
```diff
@@ -62,6 +62,7 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
 
             IEnumerable<AssignmentLinkDto> assignmentLinkDtos = sameAssignmentLinks
                 .OrderBy(linkEntity => linkEntity.Timestamp.Ticks)
+                .Where(linkEntity => Guid.TryParse(linkEntity.RowKey, out Guid result))
```
Does it make sense to move this into `_mapper`? That way it should help in case newer entry points for LinkEntity -> LinkDto conversion popup in future.

maleekha replied:
I have made some changes in this regard: essentially, the mapper while converting `LinkEntity` to `LinkDto` will map the Id only if it can be parsed to a `Guid`, else it will keep it as the default `Guid.Empty` value. 

On the client-side, I have added a check for filtering such links with `Guid.Empty` Ids, indicating that these are not "well-formed". Although the same holds here also: the check might have to be propagated in the client-side if a similar condition arises somewhere else. But given the store can be re-used where ever we want, in future, it is less likely we will have to rewrite such conditions. Any thoughts?

adbindal reviewed (changes requested):
The changes seem fine with me. I've shared some queries and thoughts relating to moving the fixes to a more common location.

maleekha commented:
I have made some changes to address the comments. In particular, I have moved the invalid-link-filtering part to the `link store`, and have also added some UI changes to reflect whenever certain links are not being rendered due to invalid-ids.

adbindal reviewed (approved):
LGTM 👍
